src/tiledb/cloud/vector_search/file_ingestion.py

```diff
@@ -0,0 +1,355 @@
+        index_uri: str,
+        file_name: Optional[str] = None,
+        config=None,
+        environment_variables: Optional[Mapping[str, str]] = None,
+        verbose: bool = False,
+        # Index creation params
+        index_type: str = "IVF_FLAT",
+        index_creation_kwargs: Optional[Dict] = None,
+        # DirectoryTextReader params
+        include: str = "*",
+        exclude: Optional[Sequence[str]] = ("[.]*", "*/[.]*"),
+        suffixes: Optional[Sequence[str]] = None,
+        max_files: Optional[int] = None,
+        text_splitter: str = "RecursiveCharacterTextSplitter",
+        text_splitter_kwargs: Optional[Dict] = None,
+        # Embedding params
+        embedding_class: str = "LangChainEmbedding",
+        embedding_kwargs: Optional[Dict] = None,
+    ) -> str:
+        """
+        Create a TileDB vector search dataset.
+        """
+        import tiledb.vector_search as vs
+        import tiledb.vector_search.embeddings as embeddings_module
+        from tiledb.vector_search.object_api import object_index
+        from tiledb.vector_search.object_readers import DirectoryTextReader
+
+        logger = get_logger_wrapper(verbose)
+        logger.debug("tiledb-vector-search=%s", vs.__version__)
+
+        # Check if the dataset already exists
+        with tiledb.scope_ctx(config):
+            if environment_variables is None:
+                environment_variables = {}
+            if index_creation_kwargs is None:
+                index_creation_kwargs = {}
+            if embedding_kwargs is None:
+                embedding_kwargs = {
+                    "dimensions": 1536,
+                    "embedding_class": "OpenAIEmbeddings",
+                    "embedding_kwargs": {
+                        "model": "text-embedding-ada-002",
+                    },
+                }
+            if text_splitter_kwargs is None:
+                text_splitter_kwargs = {
+                    "chunk_size": 500,
+                    "chunk_overlap": 50,
+                }
+
+            reader = DirectoryTextReader(
+                search_uri=file_dir_uri,
+                include=file_name if file_name is not None else include,
+                exclude=exclude,
+                suffixes=suffixes,
+                max_files=max_files,
+                text_splitter=text_splitter,
+                text_splitter_kwargs=text_splitter_kwargs,
+            )
+
+            embedding_class_ = getattr(embeddings_module, embedding_class)
+            embedding = embedding_class_(**embedding_kwargs)
+
+            if tiledb.object_type(index_uri) == "group":
+                logger.info("Existing dataset: %r. Updating reader.", index_uri)
+                index = object_index.ObjectIndex(
+                    uri=index_uri,
+                    environment_variables=environment_variables,
+                    load_embedding=False,
+                    load_metadata_in_memory=False,
+                    memory_budget=1,
```

Comment:
Why is this set to 1? Please add inline code comments. There should be a decent amount of comments explaining the purpose of values set such as this one. The goal is for others to be able to read the code + comments and understand the code and be able to work on it. If request.

Reply:
Added comment for this, this avoids loading vector data in memory since we don't want to perform queries.